core-blocks/embed/index.js

```diff
@@ -87,44 +82,88 @@ function getEmbedBlockSettings( { title, description, icon, category = 'embed',
 			}
 		},
 
-		edit: class extends Component {
+		edit: compose(
+			withSelect( ( select, ownProps ) => {
+				const { url } = ownProps.attributes;
+				// Preview is undefined if we don't know the status of it, false if it failed,
+				// otherwise it will be an object containing the embed response.
+				const preview = url ? select( 'core' ).getEmbedPreview( url ) : undefined;
+				return {
+					preview,
+				};
+			} )
+		)( class extends Component {
 			constructor() {
 				super( ...arguments );
 
-				this.doServerSideRender = this.doServerSideRender.bind( this );
+				this.setUrl = this.setUrl.bind( this );
+				this.processPreview = this.processPreview.bind( this );
 
 				this.state = {
 					html: '',
 					type: '',
 					error: false,
 					fetching: false,
 					providerName: '',
+					url: '',
 				};
 			}
 
-			componentDidMount() {
-				this.doServerSideRender();
+			componentWillMount() {
```

Comment:
`componentWillMount` is a deprecated lifecycle function, and can usually be substituted with `constructor`. Probably sensible anyways to set initial state via `this.state`.

https://reactjs.org/docs/react-component.html#the-component-lifecycle